submissions/HannaSyn/oop-exercise/index.js

```diff
@@ -0,0 +1,88 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor ( name, gender, friends, say, biologicalSpecies ) {
+      this.name = name;
+      this.biologicalSpecies = biologicalSpecies;
+      this.gender = gender;
+      this.friends = friends;
+      this.say = say
+   }
+   speak() {
+      return `${this.say} I'm ${this.biologicalSpecies}! My name is ${this.name}, my gender is ${this.gender}, and I have friends: ${this.friends}.`;
+   }
+}
+
+class Human extends Inhabitant {
+   constructor( name, gender, friends, say, legs, hands ) {
+      super( name, gender, friends, say, 'human');
+      this.legs = legs;
+      this.hands = hands;
+   } 
+   speak() {
+      return super.speak() + ` I have ${this.legs} legs and ${this.hands} hands.`
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor( name, gender, friends, say, biologicalSpecies, paws) {
+      super( name, gender, friends, say, biologicalSpecies )
+      this.paws = paws;
+   }
+   speak() {
+      return super.speak() + ` I have ${this.paws} paws.`
+   }
+}
+
+class Cat extends Animal {
+   constructor( name, gender, friends, say, paws) {
```

Comment:
How about default values for `paws` and e.g. `say`?
Same might apply for other animals.
Humans might have default hands and legs count.
This approach would allow a developer to define non-standard counts of limbs at construction time, hopefully saving their effort.
This might require you to reconsider order of arguments for dogs.